File: java-repository/games/3d.jengine/engine/game/Game.java
package game;
/*
Copyright (C) 2012 Johan Ceuppens

This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.

This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
*/
import java.awt.*;
import javax.swing.*;

import java.awt.BasicStroke;
import java.awt.Color;
import java.awt.Dimension;
import java.awt.Event;
import java.awt.Font;
import java.awt.FontMetrics;
import java.awt.GraphicsEnvironment;
import java.awt.Graphics;
import java.awt.Graphics2D;
import java.awt.Image;
import java.awt.Toolkit;
import java.awt.RenderingHints;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;

import javax.swing.ImageIcon;
import javax.swing.JPanel;
import javax.swing.Timer;
import javax.imageio.ImageIO;

import java.util.*;
import java.util.Random;
import java.awt.image.*;
import java.awt.image.BufferedImage;

public class Game extends JPanel implements ActionListener {
private String fileprefix = "./pics/";
private String musicfileprefix = "./music/";
private int SCREENWIDTH = 320;
private int SCREENHEIGHT = 200;
//private Texturef tex = new Texturef("door-48x48-1.png");
private Sphere sphere = new Sphere(100.0f,20);
private Translatef translatef = new Translatef(0,0,0);
private Camera cam = new Camera();

public Game() {
	
	Timer timer;

        addKeyListener(new TAdapter());
        setFocusable(true);
	
        setBackground(Color.white);
        setDoubleBuffered(true);

        timer = new Timer(40, this);
        timer.start();

	loadlevel1();

    }

    public void addNotify() {
        super.addNotify();
        GameInit();
    }

    public void loadlevel1()
    {
    }

    public void loadlevel2()
    {
    }

    public void GameInit() {
        LevelInit();
    }


    public void LevelInit() {

    }

/*
 * Collision detection code
 */
/****
   public boolean DoGatewayCollision()
    {
	for (int i = 0; i < gateways.size(); i++) {
		Object o = gateways.get(i);
		Gateway bo = (Gateway)o;
		if (collision(player.getx()-map.getx(),player.gety()-map.gety(),player.getw(),player.geth(),bo.getx(),bo.gety(),bo.getw(),bo.geth())) {

			currentgateway = bo;

			return true;
		}
	}
	return false;
    }	
*****/

/*
 * Drawing
 */ 
/*****    public void DrawGateways(Graphics2D g2d)
    {
	for (int i = 0; i < gateways.size(); i++) {
		Object o = gateways.get(i);
		Gateway bo = (Gateway)o;
		g2d.drawImage(bo.getImage(), bo.getx()+map.getx(), bo.gety()+map.gety(), this);
	}
    }	
*****/
	
    public void paint(Graphics g)
    {
      Graphics2D g2d = (Graphics2D) g;
	g2d.setColor(Color.black);
      g2d.fillRect(0, 0, 320, 200);
      g2d.setColor(Color.red);

	for (int i = 0; i < sphere.size(); i++) {

		Vector3 v = sphere.getVertex(i);
		//v = new Quaternion(2f,2f,2f,2f).rotateVector(v);	

		cam.movey(2f);
		//cam.rotatex(0.2f);
		v = new Vector3(cam.x(),cam.y(),cam.z());
      		g2d.fillRect((int)v.x()+(int)sphere.x(),(int)v.y()+(int)sphere.y(),1,1);
			

	} 

      //g2d.drawImage(tex.getImage(),null,null);
      Toolkit.getDefaultToolkit().sync();
      g.dispose();

    }

    class TAdapter extends KeyAdapter {
        public void keyReleased(KeyEvent e) {
	} 
      
        public void keyPressed(KeyEvent e) {

          int key = e.getKeyCode();

	   	if (key == KeyEvent.VK_LEFT) {
			sphere.setx(sphere.x()-1f);
			repaint();	
		}
	   	if (key == KeyEvent.VK_RIGHT) {
			sphere.setx(sphere.x()+1f);
			repaint();	
	   	}
	   	if (key == KeyEvent.VK_UP) {
			sphere.sety(sphere.y()-1f);
			repaint();	
	   	}
	   	if (key == KeyEvent.VK_DOWN) {
			sphere.sety(sphere.y()+1f);
			repaint();	
		}	
	   	if (key == KeyEvent.VK_X) {
	   	}
	   	if (key == KeyEvent.VK_Z) {//go back to history of talkmodes
	   	}
		if (key == KeyEvent.VK_ESCAPE) {
	   		System.exit(99);
		}
	}
    }

    public void actionPerformed(ActionEvent e) {
        repaint();  
    }

}

File: perl-repository/perlgames/perlsdlplatformgame/src/Game.pl
# Copyright (C) 2012 Johan Ceuppens

# This program is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation; either version 2 of the License, or (at your option) any later version.

# This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.

# You should have received a copy of the GNU General Public License along with this program; if not, write to the Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
#


use SDL;
use SDL::Video;
use SDL::Surface;
use SDL::Image;
use SDL::Rect;
use SDLx::App;

use lib './src/';
use Player;
use Map;
use BoundingBox;
use Gateway;


BEGIN{ unshift(@INC,"../blib/");}
BEGIN{ unshift(@INC,"../blib/lib");}
BEGIN{ unshift(@INC,"../blib/arch");}
my $player = Player->new(100,100);
my $map = undef;
my @boundingboxes = ();
my @gateways = ();

sub loadlevel1000 {
	$map = Map->new(0,0,800,600);
	$map->addImage("./pics/door-48x48-1.png");
	$mappngsurface = $map->getImage();
	push(@boundingboxes,BoundingBox->new(0,400,800,20));	
	$gw = Gateway->new(100,400,800,20,1001);
	$gw->addImage("./pics/door-48x48-1.png");
	push(@gateways, $gw);
		
};

my $screenwidth = 800;
my $screenheight = 600;

my $app = SDLx::App->new($screenwidth, $screenheight);

my $playerpngsurface;
$playerpngsurface = $player->getStaticImage();
my $mappngsurface;


SDL::Image::init(IMG_INIT_PNG);
loadlevel1000();

$app->add_show_handler( sub {
	if ( $playerpngsurface ) {
		$playerpngsurface->blit($app, undef, [$player->getx(),$player->gety(),0,0]);
	}	
	if ( $mappngsurface ) {
		$mappngsurface->blit($app, undef, [$map->getx(),$map->gety(),$map->getw(),$map->geth() ]);
	}
	foreach (@gateways) {
		$_->getImage()->blit($app, undef, [$_->getx(),$_->gety(),$_->getw(),$_->geth() ]);
	}	
	foreach (@gateways) {	
		if ( $_->collision($player) ) {
			my $k = $_->getlevelnumber();
			if ($k == 1000) {
				loadlevelnumber1000();
			} elsif ($k == 1001) {
				###loadlevelnumber1001();
			} else {	
			}
			$app->update();
			return;
		}
	}
	foreach (@boundingboxes) {	
		if ( $_->collision($player) ) {
			return;###FIXME
		}
	}
	$player->fall();
	$app->update();
});

$app->add_event_handler(
sub {
###FIXME	$app->stop();
});

$app->add_event_handler(
sub {
	my $event = shift;

	$app->draw_rect( [ 0,0,$app->w,$app->h],0);	

	if ($event->type == SDL_KEYDOWN )
	{
		my $key_name = SDL::Events::get_key_name($event->key_sym);

		$playerpngsurface = $player->getImage();
		$player->moveleft() if $key_name =~ /^j$/;
		$player->moveright() if $key_name =~ /^l$/;

	} 
	elsif ($event->type == SDL_KEYUP )
	{
		$playerpngsurface = $player->getStaticImage();
		my $key_name = SDL::Events::get_key_name($event->key_sym);
	}
	$app->update();

});

$app->run();

print "Quitting.";
